app/src/main/java/org/olpc/xorobothash/MainActivity.java

```diff
@@ -0,0 +1,164 @@
+package org.olpc.xorobothash;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.os.AsyncTask;
+import android.support.v7.app.ActionBarActivity;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ImageView;
+import android.util.Log;
+import android.widget.TextView;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.net.HttpURLConnection;
+
+
+
+public class MainActivity extends ActionBarActivity {
+
+    ImageView robotImageView;
+    String robotImagePath;
```

Comment:
Make all fields private, almost always. Sometimes for "struct" type objects you might have `public final` fields.